plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/TemplateListPage.java

```diff
@@ -785,6 +786,13 @@ public IStatus runInUIThread(IProgressMonitor monitor) {
 						}
 					}).build();
 			WizardUtils.runInWizard(jobs, getContainer());
+			if(closeAfter[0]) {
```

Comment:
No, other jobs go in ui thread and should be completed before this method continues.